app/lib/actions/katello/host/package_group/install.rb

```diff
@@ -4,6 +4,7 @@ module Host
       module PackageGroup
         class Install < Actions::EntryAction
           include Helpers::Presenter
+          middleware.use Actions::Middleware::KeepCurrentUser
```

Comment:
this keepcurrentuser middleware shouldn't be needed any more, as they are automatically used on every action

Reply:
this would be nice. Unfortunately, I ca not find the line in the code in which the KeepCurrentUser middleware is added automatically. Just want to make sure that the user is the correct one.

Reply:
Here's the commit that added it: https://github.com/theforeman/foreman-tasks/commit/cf5c3a8b9fbe5f0b853341531d023e066af88c4f

Reply:
Tried it again without 'middleware.use Actions::Middleware::KeepCurrentUser' - in this case, no audit log entry is written.

Reply:
strange, i will play with it, testing the pr now

Reply:
it seems to work for me, are you using a production environment or a dev env?  What version of foreman-tasks is installed?

Reply:
Production. Foreman 1.18.2 / Katello 3.7.1.1 / Foreman Tasks / foreman-tasks-core 0.2.5
foreman-tasks 0.13.4

Reply:
I'm OK to remove the middleware.use line if you confirm that it works. However, the middleware.use line is in a lot of other katello actions, too. Maybe it would be saver to remove the middleware.use for all katello actions as a separate task.

Reply:
ah that's why, it was added in foreman-taks 0.14.1

Reply:
yeah, lets remove it.  we need to remove it from all the others at some point

Reply:
> ah that's why, it was added in foreman-taks 0.14.1

I had a look at the 0.13.4 - in this version the change does already exist. I compared it with the files of my system and can confirm, that the change of foreman task is already in 0.13.4

Reply:
@sbernhard very strange, i cannot explain why it woudln't work then.  I can take a look on a production system as well.

Reply:
The difference in the foreman task is, that in case middleware.use line is active, the finalize input data contains: "current_user_id: 7". Without the middleware.use line the current_user_id is missing.

Reply:
Conversation from irc:
```
 jsherrill | sbernhard[m]1: hey, so i tested on 3.7 and you are right, that line is needed, however its not needed in master     
           | (based on my tests)                                                                                                 
 jsherrill | i'm not sure what changed in foreman-tasks between then and now                                                     
  aruzicka | jsherrill: hi, what line?                                                                                           
 jsherrill | aruzicka: https://github.com/theforeman/foreman-tasks/commit/cf5c3a8b9fbe5f0b853341531d023e066af88c4f               
 jsherrill | that global 'keepcurrentuser'  doesn't seem to be working on katello 3.7                                            
  aruzicka | uhmm, what's your dynflow version?                                                                                  
 jsherrill | tfm-rubygem-dynflow-1.1.0-1.el7.noarch                                                                              
 jsherrill | aruzicka: its not a huge deal, as this is around a pr going into master, but sbernhard[m]1 was testing on 3.7       
  aruzicka | seems we missed a dependency. that needs dynflow-1.1.1 to work                                                      
 jsherrill | gotcha                                                                                                              
 jsherrill | thanks aruzicka                                                                                                     
 jsherrill | i don't think we need to worry about it                                                                             
 jsherrill | but it explains the mystery                                                                                         
```